source/TradingPanel.cpp

```diff
@@ -236,35 +236,38 @@ bool TradingPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command,
 		Buy(1000000000);
 	else if(key == 'S' || (key == 's' && (mod & KMOD_SHIFT)))
 	{
-		for(const auto &it : GameData::Commodities())
+		for(const auto &it : player.Cargo().Commodities())
 		{
-			int64_t amount = player.Cargo().Get(it.name);
-			int64_t price = system.Trade(it.name);
+			const auto commodity = it.first;
+			const auto amount = it.second;
+			int64_t price = system.Trade(commodity);
 			if(!price || !amount)
 				continue;
 
-			int64_t basis = player.GetBasis(it.name, -amount);
-			player.AdjustBasis(it.name, basis);
+			int64_t basis = player.GetBasis(commodity, -amount);
 			profit += amount * price + basis;
 			tonsSold += amount;
 
-			player.Cargo().Remove(it.name, amount);
+			GameData::AddPurchase(system, commodity, -amount);
+			player.AdjustBasis(commodity, basis);
 			player.Accounts().AddCredits(amount * price);
-			GameData::AddPurchase(system, it.name, -amount);
+			player.Cargo().Remove(commodity, amount);
 		}
 		int day = player.GetDate().DaysSinceEpoch();
 		for(const auto &it : player.Cargo().Outfits())
 		{
-			if(it.first->Get("minable") <= 0. && !sellOutfits)
+			const auto outfit = it.first;
+			const auto amount = it.second;
```

Comment:
```suggestion
			const Outfit *const outfit = it.first;
			const int64_t &amount = it.second;
```

Reply:
Fair enough!  Any reason not to have outfit also be a reference?

Reply:
```c++
const Outfit &outfit = it.first;
```
Won't work, `it.first` is a `const Outfit *`.
```c++
const auto &outfit = it.first;
```
Looks like it might create a `const Outfit &` (const reference to an `Outfit`) but it's actually creating a const reference to the pointer, a `const Outfit * const`, which is very different. If it were the former, we wouldn't dereference it to access its members, but we do have to dereference it. I think it just looks confusing.

Also, on the other line, we do `using namespace std;` in .cpp files and don't generally include the `std::` prefix on members of that namespace here, only in .h files.

Reply:
Whoops, there's a reflex for you - I hadn't even noticed I'd added std::.

For the other, I mean const Outfit * const &outfit.  If you don't like the extra &, I can take it out easily enough.  Optimization should mean both ways result in the same code, anyway.

Reply:
Usually types like ints and pointers are passed by value because going through the reference is a tad more expensive than simply copying the value. It doesn't really make a difference here though, so I don't mind either way.